sirepo/package_data/static/js/controls.js

```diff
@@ -582,16 +582,74 @@ SIREPO.viewLogic('quadrupoleView', function(appState, controlsService, panelStat
     };
 });
 
+SIREPO.app.directive('optimizationPicker', function(latticeService) {
+    return {
+        restrict: 'A',
+        scope: {
+            controller: '='
+        },
+        template: [
+                  `<div>
+                     <div class="container-fluid">
+                       <div class="row" data-ng-show="::showTabs">'
+                         <div class="col-sm-12">
+                           <ul class="nav nav-tabs">
+                             <li role="presentation" data-ng-class="{active: activeTab == 'targets'}"><a href data-ng-click="activeTab = 'targets'">Targets</a></li>
+                             <li role="presentation" data-ng-class="{active: activeTab == 'inputs'}"><a href data-ng-click="activeTab = 'inputs'">Inputs</a></li>
+                           </ul>
+                         </div>
+                       </div>
+                       <br />
+                       <div data-ng-if="activeTab == 'targets'" class="row">
+                         <div class="clearfix" data-optimizer-table=""></div>
+                       </div>
+                       <div data-ng-if="activeTab == 'inputs'" class="row">
+                         <div class="container-fluid">
+                         <form name="form">
+                           <table ng-repeat="(inputType, inputs) in appState.models.optimizerSettings.inputs" style="float: left; margin: 1em;">
+                             <thead>
+                               <th>{{capitalizeFirstLetter(inputType)}}</th>
```

Comment:
You can use `stringsService.ucFirst` (stringsService is in sirepo.js). See examples of it in action in sirepo-components.js